### PR TITLE
Update chromecast wifi gather module to use Scanner for scanning in bulk

### PR DESCRIPTION
**Pull request by jhart-r7**
This simply updates this module to use `Scanner` so that one can run this across larger networks more easily.

## Verification

- [x] Start `msfconsole`
- [x] `use auxiliary/gather/chromecast_wifi`
- [x] Set `RHOSTS` to some network with 1 or more chromecasts
- [x] **Verify** that each chromecast has a unique table printed out with the wireless networks it sees

```
msf auxiliary(chromecast_wifi) > run

Wireless Access Points from 192.168.1.57
========================================

BSSID              PWR  ENC   CIPHER  AUTH  ESSID
-----              ---  ---   ------  ----  -----
00:2a:a8:f7:ff:93  -45  WPA2  CCMP    PSK   blah (*)

Wireless Access Points from 192.168.1.50
========================================

BSSID              PWR  ENC   CIPHER  AUTH  ESSID
-----              ---  ---   ------  ----  -----
00:2a:a8:f7:ff:93  -45  WPA2  CCMP    PSK   blah (*)
01:17:1d:b7:e0:78  -86  WPA2  CCMP    PSK   blah2


[*] Scanned 2 of 2 hosts (100% complete)
```

### CODE REVIEW

**Comment by h00die** [comment]
I have a few chromecasts at home, I wouldn't mind taking this when its ready.  Maybe i'll throw together some docs for it as well.  Just let me know when Travis has been appeased w/ sacrificed goats

**Comment by jhart-r7** [comment]
Thanks @h00die.  My goats seem to be inadequate -- I can't get travis to pass and the failure seems entirely unrelated to this PR.

**Comment by h00die** [comment]
Awesome.  I'll check it over tonight or tomorrow, write up some docs, and get this rolling.

**Comment by wvu-r7** [comment]
Why was this module ```aux/gather``` in the first place? That's dumb.

File: modules/auxiliary/gather/chromecast_wifi.rb [hunk 1]
```diff
@@ -24,16 +25,16 @@ def initialize(info = {})
...
-  def run
+  def run_host(ip)
     res = scan
 
     return unless res && res.code == 200
 
-    waps = Rex::Text::Table.new(
-      'Header' => 'Wireless Access Points',
+    waps_table = Rex::Text::Table.new(
+      'Header' => "Wireless Access Points from #{rhost}",
```
Comment (wvu-r7) [inline modules/auxiliary/gather/chromecast_wifi.rb]: Change ```rhost``` to ```ip```. Thanks.

**Comment by wvu-r7** [comment]
I think I remember now why it went in ```aux/gather``` instead of ```aux/scanner```.

```
wvu@kharak:~/metasploit-framework:master$ search chromecast
./modules/auxiliary/admin/chromecast
./modules/auxiliary/admin/chromecast/chromecast_reset.rb
./modules/auxiliary/admin/chromecast/chromecast_youtube.rb
./modules/auxiliary/gather/chromecast_wifi.rb
./modules/auxiliary/scanner/http/chromecast_webserver.rb
wvu@kharak:~/metasploit-framework:master$ 
```

It was meant to be used in conjunction with ```auxiliary/scanner/http/chromecast_webserver``` *or* multicast recon. I like your way better, @jhart-r7, since it provides a better "site survey" of WAPs in the area.

File: modules/auxiliary/gather/chromecast_wifi.rb [hunk 2]
```diff
@@ -56,15 +57,16 @@ def run
...
-      :host => rhost,
-      :port => rport,
-      :proto => 'tcp',
-      :type => 'chromecast.wifi',
-      :data => waps.to_csv
-    )
+    unless waps_table.rows.empty?
+      print_line(waps_table.to_s)
+      report_note(
+        :host => rhost,
```
Comment (wvu-r7) [inline modules/auxiliary/gather/chromecast_wifi.rb]: Oh, this should have been ```ip```, too.

**Comment by wvu-r7** [comment]
Quick PR for you, @jhart-r7. Cheers.

**Comment by wvu-r7** [comment]
Now if only we could make ```chromecast_youtube``` an ```aux/scanner```... /s

**Comment by busterb** [comment]
Ritual goat sacrifice in #7748 and now on master

**Comment by h00die** [comment]
When setting it to a class C, if it hits a host w/o port 8008 it kills the entire run instead of just the thread/IP.
```
msf auxiliary(chromecast_wifi) > show options

Module options (auxiliary/gather/chromecast_wifi):

   Name     Current Setting  Required  Description
   ----     ---------------  --------  -----------
   Proxies                   no        A proxy chain of format type:host:port[,type:host:port][...]
   RHOSTS   10.10.10.0/24   yes       The target address range or CIDR identifier
   RPORT    8008             yes       The target port
   SSL      false            no        Negotiate SSL/TLS for outgoing connections
   THREADS  20               yes       The number of concurrent threads
   VHOST                     no        HTTP server virtual host

msf auxiliary(chromecast_wifi) > run

[-] Auxiliary failed: RuntimeError unreachable: The host (10.10.10.0:8008) was unreachable.
[-] Call stack:
[-]   /metasploit-framework/lib/msf/core/module.rb:290:in `fail_with'
[-]   /metasploit-framework/modules/auxiliary/gather/chromecast_wifi.rb:86:in `rescue in scan'
[-]   /metasploit-framework/modules/auxiliary/gather/chromecast_wifi.rb:89:in `scan'
[-]   /metasploit-framework/modules/auxiliary/gather/chromecast_wifi.rb:32:in `run_host'
[-]   /metasploit-framework/lib/msf/core/auxiliary/scanner.rb:135:in `block (2 levels) in run'
[-]   /metasploit-framework/lib/msf/core/thread_manager.rb:100:in `block in spawn'
[*] Auxiliary module execution completed
```

**Comment by h00die** [comment]
I guess that you could argue thats the difference between a scanner and a gather module, however, I want to ensure that's the behavior we're looking for

**Comment by wvu-r7** [comment]
@h00die: That's part of what my patch fixed. Please try again now that is has merged. :)

**Comment by wvu-r7** [comment]
@h00die: That's part of what my patch fixed. Please try again now that it has merged. :)

**Comment by h00die** [comment]
looks good to me, landing.
```
msf > use auxiliary/scanner/http/chromecast_wifi 
msf auxiliary(chromecast_wifi) > set rhosts 10.10.10.0/24
rhosts => 10.10.10.0/24
msf auxiliary(chromecast_wifi) > set threads 20
threads => 20
msf auxiliary(chromecast_wifi) > set verbose true
verbose => true
msf auxiliary(chromecast_wifi) > run

Wireless Access Points from 10.10.10.11
========================================

BSSID              PWR  ENC   CIPHER  AUTH  ESSID
-----              ---  ---   ------  ----  -----
00:11:22:33:44:55  -59  WPA2  CCMP    PSK   Rapid7 (*)
aa:11:22:33:44:66  -71  OPN                 xfinitywifi

[*] Scanned  26 of 256 hosts (10% complete)
[*] Scanned  53 of 256 hosts (20% complete)
[*] Scanned  79 of 256 hosts (30% complete)
[*] Scanned 105 of 256 hosts (41% complete)
[*] Scanned 129 of 256 hosts (50% complete)
[*] Scanned 154 of 256 hosts (60% complete)
Wireless Access Points from 10.10.10.12
=========================================

BSSID              PWR  ENC   CIPHER  AUTH  ESSID
-----              ---  ---   ------  ----  -----
bb:aa:22:33:44:66  -94  WPA   TKIP    PSK   wifi
bb:aa:cc:dd:44:66  -54  WPA2  CCMP    PSK   wifi2 (*)

[*] Scanned 180 of 256 hosts (70% complete)
Wireless Access Points from 10.10.10.16
=========================================

BSSID              PWR  ENC   CIPHER  AUTH  ESSID
-----              ---  ---   ------  ----  -----
bb:aa:cc:dd:44:66  -54  WPA2  CCMP    PSK   wifi2 (*)

[*] Scanned 222 of 256 hosts (86% complete)
Wireless Access Points from 10.10.10.23
=========================================

BSSID              PWR  ENC   CIPHER  AUTH  ESSID
-----              ---  ---   ------  ----  -----
bb:aa:cc:dd:44:66  -63  WPA2  CCMP    PSK   wifi2 (*)
00:11:22:33:44:55  -85  WPA2  CCMP    PSK   Rapid7 (*)

[*] Scanned 241 of 256 hosts (94% complete)
[*] Scanned 256 of 256 hosts (100% complete)
[*] Auxiliary module execution completed
```

**Comment by h00die** [comment]
## Release Notes

Update to the chromecast_wifi module to now use Scanner and reside as a scanner instead of gather.

**Comment by wvu-r7** [comment]
Thank you, @jhart-r7 && @h00die. Happy holidays!

**Pull request merged**